apis/openstack-swift/src/main/java/org/jclouds/openstack/swift/v1/blobstore/RegionScopedSwiftBlobStore.java

```diff
@@ -188,11 +197,17 @@ public boolean createContainerInLocation(Location location, String container) {
 
    @Override
    public boolean createContainerInLocation(Location location, String container, CreateContainerOptions options) {
+      boolean resultContainerCreate = false;
       checkArgument(location == null || location.equals(region), "location must be null or %s", region);
       if (options.isPublicRead()) {
-         return api.getContainerApi(regionId).create(container, ANYBODY_READ);
+         resultContainerCreate = api.getContainerApi(regionId).create(container, ANYBODY_READ);
       }
-      return api.getContainerApi(regionId).create(container, BASIC_CONTAINER);
+      resultContainerCreate = api.getContainerApi(regionId).create(container, BASIC_CONTAINER);
+      if (resultContainerCreate) {
+         Container val = api.getContainerApi(regionId).get(container);
+         containerCache.put(container, Optional.fromNullable(val));
+      }
+      return resultContainerCreate;
```

Comment:
How is creating a container related to dynamic large objects?  If this is a separate change, submit a separate pull request.